adcm_aio_client/core/objects/cm.py

```diff
@@ -117,13 +124,41 @@ class ServicesNode(PaginatedChildAccessor[Cluster, Service, None]):
 
 
 class Component(InteractiveChildObject[Service]):
-    @property
-    def id(self: Self) -> int:
-        return int(self._data["id"])
 
     def get_own_path(self: Self) -> Endpoint:
         return (*self._parent.get_own_path(), "components", self.id)
 
 
 class ComponentsNode(NonPaginatedChildAccessor[Service, Component, None]):
     class_type = Component
+
+
+class HostProvider(Deletable, InteractiveObject):
```

Comment:
You can inherit from `RootInteractiveObject` instead of `InteractiveObject` and specify PATH  instead of implementing `get_own_path`